User/UserRepository.php

```diff
@@ -20,4 +20,38 @@ public function findByUsername($username);
      * @return bool
      */
     public function supportsClass($class);
+
+    /**
+     * Add User to the repository.
+     *
+     * @param User $user
+     * @param bool $save
+     * @return void
+     */
+    public function add(User $user, $save = true);
+
+    /**
+     * Flush the repository.
+     *
+     * @param User|null $user
+     * @return void
+     */
+    public function save(User $user = null);
```

Comment:
je interface heeft niets met doctrine te maken. Wie zegt er dat je je files met doctrine gaat opslaan? Misschien gaat ge die wel in een file opslaan of door iemand laten opschrijven op papier?

Een repository zou eigenlijk gewoon moeten werken zoals een "collectie van items". Hoe doctrine/databases/filesystem/... werken zijn pure implementatie details.

bijhorend leeswerk: http://shawnmc.cool/the-repository-pattern